agent/agent.go

```diff
@@ -2,77 +2,77 @@ package agent
-
 	"github.com/libp2p/go-libp2p"
 	"github.com/libp2p/go-libp2p-core/host"
 	"github.com/libp2p/go-libp2p-core/peer"
 	pubsub "github.com/libp2p/go-libp2p-pubsub"
 	configLibp2p "github.com/libp2p/go-libp2p/config"
-	"github.com/multiformats/go-multiaddr"
+	"github.com/maticnetwork/libp2p-gossip-bench/network"
+	ma "github.com/multiformats/go-multiaddr"
 )
 
-//go:embed latencies.json
-var latencyDataRaw string
+const (
+	// overlay parameters
+	gossipSubD   = 8  // topic stable mesh target count
+	gossipSubDlo = 6  // topic stable mesh low watermark
+	gossipSubDhi = 12 // topic stable mesh high watermark
+
+	// gossip parameters
+	gossipSubMcacheLen    = 6   // number of windows to retain full messages in cache for `IWANT` responses
+	gossipSubMcacheGossip = 3   // number of windows to gossip about
+	gossipSubSeenTTL      = 550 // number of heartbeat intervals to retain message IDs
+
+	// fanout ttl
+	// gossipSubFanoutTTL = 60000000000 // TTL for fanout maps for topics we are not subscribed to but have published to, in nano seconds
+
+	// heartbeat interval
+	gossipSubHeartbeatInterval = 700 * time.Millisecond // frequency of heartbeat, milliseconds
+
+	// topic for pubsub
+	topicName = "Topic"
+)
 
 type Agent struct {
-	host      host.Host
-	logger    *log.Logger
-	gossipSub *pubsub.PubSub
-	config    *Config
-	topic     *pubsub.Topic
-	latency   *LatencyData
+	Host   host.Host
+	Logger *log.Logger
+	Config *AgentConfig
+	Topic  *pubsub.Topic
 }
 
-type Config struct {
-	Addr             *net.TCPAddr
-	HttpAddr         *net.TCPAddr
-	RendezvousString string
-	City             string
-	ID               string
-	MaxPeers         int64
-	MinInterval      int64
-	Transport        configLibp2p.TptC
-}
+var _ network.ClusterAgent = &Agent{}
 
-func DefaultConfig() *Config {
-	c := &Config{
-		//Addr:     &net.TCPAddr{IP: net.ParseIP("127.0.0.1"), Port: 6000},
-		//HttpAddr: &net.TCPAddr{IP: net.ParseIP("0.0.0.0"), Port: 7000},
-	}
-	return c
+type AgentConfig struct {
+	Transport     configLibp2p.TptC
+	MsgReceivedFn network.MsgReceived
 }
 
 // pubsubQueueSize is the size that we assign to our validation queue and outbound message queue for
 // gossipsub.
 const pubsubQueueSize = 600
 
-func NewAgent(logger *log.Logger, config *Config) (*Agent, error) {
-	listenAddr, err := multiaddr.NewMultiaddr(fmt.Sprintf("/ip4/%s/tcp/%d", config.Addr.IP.String(), config.Addr.Port))
+func NewAgent(logger *log.Logger, config *AgentConfig) *Agent {
+	return &Agent{
+		Logger: logger,
+		Config: config,
+	}
+}
+
+func (a *Agent) Listen(ipString string, port int) error {
+	listenAddr, err := ma.NewMultiaddr(fmt.Sprintf("/ip4/%s/tcp/%d", ipString, port))
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	addrsFactory := func(addrs []multiaddr.Multiaddr) []multiaddr.Multiaddr {
-		addr, _ := multiaddr.NewMultiaddr(fmt.Sprintf("/ip4/%s/tcp/%d", config.Addr.IP.String(), config.Addr.Port))
+	addrsFactory := func(addrs []ma.Multiaddr) []ma.Multiaddr {
```

Comment:
I do not think this is needed anymore (please double check) since it was only required to support DNS.

Reply:
Removed